### ISSUE TITLE
Custom application instead of getty / login

### ISSUE THREAD

**Issue by winkmichael** [troglobit/finit#116]
Cool cool project! Really loving it so nice and clean

I am trying to run my own program for the login after init.
We do this via inittab like so

1:2345:respawn:/sbin/getty 38400 tty1 -i -n -l /opt/logintool

I can't seem to see anyway to achieve this type of functionality.

Is there any way to run our own program like we do via the inittab?

Thank you very much!

**Comment by troglobit** [comment]
Thanks for the kind words! :smiley: 

So Finit doesn't have inittab, and this particular feature of starting a service after someone has logged in is not available at the moment. It is however a good idea!

Starting services based on *conditions* is something in Finit is done using the following syntax:

    service <COND> [RUNLEVEL] CMD ARGS -- DESC

In you case you may not want to run a complete service (daemon), but rather start a background script.  This is called one-shot task:

    task <COND> [RUNLEVEL] CMD ARGS -- DESC

So, with existing conditions I could start a script when the default route is installed, like this:

    task <net/route/default> /bin/myscript.sh -- ET Phone Home

Currently supported conditions are [listed here](https://github.com/troglobit/finit/blob/master/docs/conditions.md#built-in-conditions).

We could treat your question as a feature reqeust if you like?  Maybe a condition like `<utmp/login>` or `<utmp/login/tty1>` could be added fairly simple?

**Comment by winkmichael** [comment]
Thank you for that very comprehensive and prompt response.

I think I might be asking my question incorrectly, so let me try again (;

From the main document page the finit.conf example, I see the option to run traditional getty

`tty [12345] /sbin/getty -L 115200 /dev/tty3 linux`

However, it looks like that mechanism (at least as how I understand it has been depreciated) with respect to login.

The /etc/finit.d/getty.conf file is like so

`> tty [12345] /dev/tty1 115200 vt100 noclear nowait`

My attempt to run traditional getting which spans (logintool) like so

`tty [12345] "/sbin/getty 38400 tty1 -i -n -l /opt/logintool"
`

I've tried variations of modifying getty.conf to run /opt/logintool with no luck

I don't need to run this process after login, but run it instead of the traditional login.

In affect logintool is my authentication system.

==

Second question and this one is probably a feature request. In inittab there is a respawn tag which will restart my /opt/logintool program if it dies, anything like that available with finit.

Thanks yet again, very much appreciate you taking the time to help me!

**Comment by troglobit** [comment]
The `tty` stanza is not deprecated. Finit has its own built-in getty these days, which is used with

    tty /dev/tty1 ....

An external getty is assumed if `/dev/` is *not* found first on the line.  But you need to skip the
quotation marks, so something like this should work:

    tty [12345] /sbin/getty 38400 tty1 -i -n -l /opt/logintool -- Optional description

Both the `tty` and `service` commands are monitored by Finit.  So when the user logs out, or
a service exits, it is automatically respawned by Finit.

Curious, why can't you use PAM?

**Comment by winkmichael** [comment]
Thank you, I would swear I tried that  [ about 3 VM burns ago... (:  ] but will give it another go! Thanks again

"logintool" uses special USB input device for disabled people - it takes various input options ranging from finger print scanning to mouse pattern movements. It does authentication when no keyboard or limited keyboard is available.

**Comment by troglobit** [comment]
I guess we can close this issue :)
